### ISSUE TITLE
Glitchy shifts in surfaces

### ISSUE THREAD

**Issue by vinamragehlot** [mikedh/trimesh#1297]
TL;DR: To accommodate for bigger geometry in a field of view, Trimesh shifts the datums, which results in a shift in the output of the project.

*Problem statement*
The aim was to project the thickness(along height, z) of a 3d part (.stl) onto the top face of that part(so if you look from top, you can know how thick the part is in a specific area, image below). 

![part](https://user-images.githubusercontent.com/26214766/126873636-b9a9de91-574f-46d4-a83c-a780363276d8.png)

*Issue*
So, the intersection of a line (along height) with the surface was noted(for all cut-sections along the y-axis), and the length of the intersected line was saved as the thickness of that area(/point) on the top surface (shown in image below). The problem is that some specific area are shifted(marked below in red) with respect to the surrounding geometry.

![thickness_output](https://user-images.githubusercontent.com/26214766/126873401-3bda6412-6c75-4b3e-ac50-ecfc67dc1181.png)

* Steps to reproduce the glitch*
*The glitch doesn't happen for simple geometries with constant cross-sections*

When each cross-section of a part (with varying cross-section) is exported as an image(plotly.go in this case), we notice that the same point(figure 2) has different coordinates. Trimesh somehow resets the datum y-axis(image) for each frame to accommodate the change in the size of the geometry, which results in the shift of the area where the bigger geometry exists.

![section](https://user-images.githubusercontent.com/26214766/126873436-28ec3228-3378-4caf-b152-ce100b3ea976.png)

*Possible solutions*
If there's a way to fix the datum of the geometry, so it doesn't shift in order to accommodate the change in geometry, the final result can be assured to be right. Somehow, the datum seems to depend on the center-of-mass of the section.

I couldn't find any solution through the documentation, or other issues (I was very thorough with my search). Any help would be much appreciated.

**Comment by vinamragehlot** [comment]
I digged a bit deeper to find out where exactly in the code was this happening, and turns out it was the affinity.rotate() function. I'll be making a separate thread about that. 
Closing this now.
